src/lazy/binary/encoded_value.rs

```diff
@@ -53,33 +52,29 @@ pub(crate) struct EncodedValue<HeaderType: EncodedHeader> {
     // and IonType.
     pub(crate) header: HeaderType,
 
-    // Each encoded value has up to five components, appearing in the following order:
+    // Each encoded value has up to four components, appearing in the following order:
     //
-    // [ field_id? | annotations? | header (type descriptor) | header_length? | value ]
+    // [ annotations? | header (type descriptor) | header_length? | value_body ]
```

Comment:
🗺️ The `field_id` offset information (and related methods) have been removed from the `EncodedValue` type and now lives in each encoding's `LazyRawFieldName` implementation. This means that we only pay the data size and offset processing cost of field IDs when we're actually processing a struct field.